server_test.go

```diff
@@ -3350,6 +3350,81 @@ func TestMaxBodySizePerRequest(t *testing.T) {
 	}
 }
 
+func TestStreamRequestBody(t *testing.T) {
```

Comment:
This tests checks if the result is correct. But it doesn't test if the body is actually being streamed or still completely read in one go.

It would be a better test to send the header then send part of the body, then wait for the `Handler` to be called. And only then send the rest of the body.

Reply:
Maybe even send it and read it in chunks with some delay in between to really test the streaming.

Reply:
Thx for the review! Will improve them soon.